lib/crypto/crypto_keys_subr.c

```diff
@@ -458,11 +458,9 @@ crypto_keys_subr_generate_RSA(RSA ** priv, RSA ** pub)
 	if (*pub != NULL)
 		RSA_free(*pub);
 
-	*priv = RSA_generate_key(2048, 65537, NULL, NULL);
-	if (*priv == NULL) {
-		warn0("%s", ERR_error_string(ERR_get_error(), NULL));
+	*priv = crypto_compat_RSA_generate_key();
+	if (*priv == NULL)
```

Comment:
Yes, I'm not sure why I didn't use the normal `if ((foo = bar()) == NULL) {` idiom here.  Maybe just because I wrote this a long time ago and hadn't settled on it at that point.